### PR TITLE
Basic Twitter OAuth implementation

### PR DESCRIPTION
**Pull request by jimmyfnugent**
Got a basic version working (thanks to @mattydoincode !). I left some TODOs, but the major things left to do:
- Create a users DB so we can associate multiple identity providers with a single Smashbits user.
- Do something with the login on the client.

### CODE REVIEW

**Comment by mattydoincode** [comment]
@jimmyfnugent it's up!

File: server/index.js [hunk 1]
```diff
@@ -1,45 +1,112 @@
...
+  helmet({
+    noCache: {
+      action: 'deny',
+    },
+  })
+);
+app.use(
+  session({
+    // TODO(thenuge): Use a better secret.
+    secret: 'kirby',
```
Comment (hopskipnfall) [inline server/index.js:43]: I asked TGA to tell me a secret:

i-think-tea-is-overrated-i-mean-its-just-leaves

File: server/index.js [hunk 2]
```diff
@@ -1,45 +1,112 @@
...
+    credentials: true, // allow session cookie from browser to pass through
+  })
+);
 
 app.get('/bits', (req, res) => {
   res.setHeader('Content-Type', 'application/json');
   getBits(req)
-      .then(result => res.send(JSON.stringify({ bits: result })))
-      .catch(error => res.status(500).send(error));
+    .then((result) => res.send(JSON.stringify({ bits: result })))
```
Comment (hopskipnfall) [inline server/index.js:71]: these extra parens are dumb

File: server/src/store.js [hunk 3]
```diff
@@ -8,34 +8,40 @@ const DEFAULT_PAGE_SIZE = 25;
...
 export function queryBits({
-    sort=SORT_PARAM_DATE,
-    offset=0,
-    limit=DEFAULT_PAGE_SIZE,
-    mainChars,
-    vsChars,
-    stages,
-    standaloneTags,
-  } = {}) {
+  sort = SORT_PARAM_DATE,
```
Comment (hopskipnfall) [inline server/src/store.js:20]: whaaaatt even is this syntax

File: server/index.js (hunk 2 above)
Comment (jimmyfnugent) [reply server/index.js:71]: Lol these formatting changes come from Matt's formatter, and I got frustrated trying to set up prettier via Sublime on Windows. I am too lazy to revert them so there.

File: server/src/store.js (hunk 3 above)
Comment (jimmyfnugent) [reply server/src/store.js:20]: It's how you do named parameters: https://javascript.info/destructuring-assignment#smart-function-parameters

**Pull request merged**